test/components/viral/system/modal_component_test.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'application_system_test_case'
+
+module System
+  class ModalComponentTest < ApplicationSystemTestCase
+    test 'with title' do
+      title = 'This is a modal title'
+      visit('rails/view_components/modal_component/default')
+      within('.Viral-Preview > [data-controller-connected="true"]') do
+        assert_text title
+      end
+    end
+
+    test 'with title and body' do
+      body = 'This is a modal body'
+      visit('rails/view_components/modal_component/default')
+      within('.Viral-Preview > [data-controller-connected="true"]') do
+        assert_selector(:xpath, './div/div/div') do
```

Comment:
**chore (blocking):** Remove the assert_selector as we are already within the component, and you are checking that the text is there

Reply:
In the [capybara](https://rubydoc.info/github/teamcapybara/capybara#beware-the-xpath--trap) documentation that was provided, it says that the .// expression means "any descendant of the current node". Yes, I'm in the component, but the component is made up of a title and body and I was trying to be more specific and look in the body for the text. 
Is this closer to what you're thinking? Please see f07000c01022cbf43440954ed68fe3bf02078196.